pkg/remoteconfig/state/products/apmsampling/remote_rates.go

```diff
@@ -27,7 +27,21 @@ type TargetTPS struct {
 	Mechanism SamplingMechanism `msgpack:"4"`
 }
 
-// APMSampling is the list of target tps
+type RareSamplerConfig string
+
+const (
+	RareSamplerConfigUnset    RareSamplerConfig = "" // default
+	RareSamplerConfigEnabled  RareSamplerConfig = "ENABLED"
+	RareSamplerConfigDisabled RareSamplerConfig = "DISABLED"
```

Comment:
Can't we just use `bool`?

How about this?
- `RareSamplerConfig` is a struct
- It has a property called `enabled` which is bool
- `APMSampling` contains `*RareSamplerConfig` instance. If it is `null`, `RareSamplerConfig` doesn't exist, so we fall back to default values

Reply:
It would totally work but I'm not a huge fan because:

 - a nullable bool is slightly less explicit than three values called `UNSET`, `ENABLED`, `DISABLED` imo 9but no big deal)
 - if `RareSamplingConfig` is a struct and if we think we'll have to add values in that struct later (see you previous comment), then we can't set only some of the fields in the struct, it's all or nothing. To achieve the same flexibility, we would have to have two nested structs IIUC

Reply:
> then we can't set only some of the fields in the struct, it's all or nothing.
Sorry I didn't get this part. Why? I thought in msgpack we can add new fields safely.

Reply:
> a nullable bool is slightly less explicit than three values called UNSET, ENABLED, DISABLED imo

Could you give some pros and cons?

Reply:
Maybe I'm missing something but consider the following example:

```
type APMSapmling struct {
    RareSamplerConfig *RareSamplerConfig
}

type RareSamplerConfig struct {
    Enabled bool
    SomeFutureNewConfig bool
}
```

How would you express "I want `SomeFutureNewConfig` to be true, but I don't want to touch the value of `Enabled`? I don't see a solution

Reply:
> a nullable bool is slightly less explicit than three values called UNSET, ENABLED, DISABLED imo

> Could you give some pros and cons?

With a nullable value, I'm not 100% sure that the `null` case is actually valid unless I read other code until I find a null check.

Does that make sense?